src/pybind/mgr/cephadm/module.py

```diff
@@ -2056,6 +2072,22 @@ def get_osd_uuid_map(self, only_up=False):
     def apply_drivegroups(self, specs: List[DriveGroupSpec]):
         return [self._apply(spec) for spec in specs]
 
+    def get_osdspec_affinity(self, osd_id: str) -> str:
+        ret, out, err = self.mon_command({
+            'prefix': 'osd metadata',
+            'id': int(osd_id),
+            'format': 'json'
+        })
+        if ret != 0:
+            self.log.warning(f"Caught error on calling 'osd metadata {osd_id}' -> {err}")
+            return ''
+        try:
+            metadata = json.loads(out)
+        except json.decoder.JSONDecodeError:
+            self.log.error(f"Could not decode json -> {out}")
+            return ''
```

Comment:
Hm. what about raising instead of silently fail here? if we're getting an invalid json, something is wrong.

Reply:
using `self.get('osd_metadata')` this won't be needed anymore.